include/sparrow/record_batch.hpp

```diff
@@ -0,0 +1,239 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <algorithm>
+#include <initializer_list>
+#include <ranges>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "sparrow/array.hpp"
+#include "sparrow/utils/contracts.hpp"
+
+namespace sparrow
+{
+    /**
+     * Table-like data structure.
+     *
+     * A record batch is a collection of equal-length arrays mapped to
+     * names.
+     *
+     * @tparm T the type of the names
+     */
+    template <class T = std::string>
```

Comment:
do we really need string as a template here?

Reply:
Probably not, I'll remove it.